catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -465,6 +474,23 @@ private void skipBytes(InputStream is, String bytesToSkip) throws IOException {
         }
     }
 
+    private void alignStream(InputStream in, long requestedBytesToSkip, long responseBytesSkipped)
+            throws IOException {
+        try {
+            if (requestedBytesToSkip > responseBytesSkipped) {
+                LOGGER.error(
+                        "Source incorrect byte range, skipping ahead [{}] bytes in the input stream",
+                        requestedBytesToSkip - responseBytesSkipped);
+                long skipped = in.skip(requestedBytesToSkip - responseBytesSkipped);
+            } else if (requestedBytesToSkip < responseBytesSkipped) {
+                throw new IOException(
```

Comment:
Ya, I can add a log message as well. 

The URLResourceReader's retrieveResource method will throw this IOException and it should be handled as a failed download attempt in the ReliableResourceDownloader, retrying if it still has retries left, or failing if the amount of retries has been passed.